src/NLog.Extensions.Logging/NLogMessageParameterList.cs

```diff
@@ -17,15 +17,15 @@ internal class NLogMessageParameterList : IList<NLog.MessageTemplates.MessageTem
 
         public NLogMessageParameterList(IReadOnlyList<KeyValuePair<string, object>> parameterList, bool includesOriginalMessage)
         {
-            List<KeyValuePair<string, object>> validParameterList = includesOriginalMessage ? null : new List<KeyValuePair<string, object>>();
+            var validParameterList = includesOriginalMessage ? null : new List<KeyValuePair<string, object>>();
             for (int i = 0; i < parameterList.Count; ++i)
             {
-                if (!string.IsNullOrEmpty(parameterList[i].Key) && (parameterList[i].Key != NLogLogger.OriginalFormatPropertyName || i == parameterList.Count - 1))
+                var paramPair = parameterList[i];
+                if (!string.IsNullOrEmpty(paramPair.Key) && (paramPair.Key != NLogLogger.OriginalFormatPropertyName || i == parameterList.Count - 1))
                 {
-                    if (validParameterList != null)
+                    if (validParameterList != null && paramPair.Key != NLogLogger.OriginalFormatPropertyName)
```

Comment:
Looks like it is matching the condition just above

Reply:
yes indeed, was also thinking to refactor that

(note I only moved the check)

Reply:
refactored :)